code/modules/awaymissions/mission_code/capture_the_flag.dm

```diff
@@ -0,0 +1,235 @@
+		dropped(user)
+		return
+	for(var/mob/M in player_list)
+		if (M.z == src.z)
+			M << "<span class='userdanger'>/The [src] has been taken!</span>"
+	SSobj.processing.Remove(src)
+
+/obj/item/weapon/twohanded/required/ctf/dropped(mob/user)
+	reset_cooldown = world.time + 200 //20 seconds
+	SSobj.processing |= src
+	for(var/mob/M in player_list)
+		if (M.z == src.z)
+			M << "<span class='userdanger'>/The [src] has been dropped!</span>"
+
+
+/obj/item/weapon/twohanded/required/ctf/red
+	name = "red flag"
+	icon_state = "banner-red"
+	item_state = "banner-red"
+	desc = "A red banner, used to play capture the flag."
+
+
+/obj/item/weapon/twohanded/required/ctf/blue
+	name = "blue flag"
+	icon_state = "banner-blue"
+	item_state = "banner-blue"
+	desc = "A blue banner, used to play capture the flag."
+
+
+
+/obj/machinery/capture_the_flag
+	name = "CTF Controller"
+	desc = "Used for running friendly games of capture the flag."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "syndbeacon"
+	var/team = WHITE_TEAM
+	var/points = 0
+	var/points_to_win = 3
+	var/list/team_members = list()
+	var/ctf_enabled = FALSE
+	var/ctf_gear = /datum/outfit/ctf
+
+/obj/machinery/capture_the_flag/red
+	team = RED_TEAM
+	ctf_gear = /datum/outfit/ctf/red
+
+/obj/machinery/capture_the_flag/blue
+	team = BLUE_TEAM
+	ctf_gear = /datum/outfit/ctf/blue
+
+/obj/machinery/capture_the_flag/attack_ghost(mob/user)
+	if(ctf_enabled == FALSE)
+		return
+	if(user.ckey in team_members)
+		if(user.timeofdeath + CTF_RESPAWN_COOLDOWN < world.time) //If we've been dead for at least 15 seconds)
+			spawn_team_member(user)
+	for(var/obj/machinery/capture_the_flag/CTF in machines)
+		if(CTF == src || CTF.ctf_enabled == FALSE)
+			continue
+		if(CTF.team_members.len > src.team_members.len)
+			user << "[src] has more team members than [CTF]. Try joining [CTF.team] to even things up."
+			return
+		if(user.ckey in CTF.team_members)
+			user << "No switching teams while the round is going!"
+	team_members |= user.ckey
+	var/client/new_team_member = user.client
+	spawn_team_member(new_team_member)
+
+/obj/machinery/capture_the_flag/proc/spawn_team_member(client/new_team_member)
+	var/mob/living/carbon/human/M = new/mob/living/carbon/human(get_turf(src))
+	new_team_member.prefs.copy_to(M)
+	M.key = new_team_member.key
+	M.faction = team
+	M.equipOutfit(ctf_gear)
+
+/obj/machinery/capture_the_flag/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/twohanded/required/ctf))
+		var/obj/item/weapon/twohanded/required/ctf/flag = I
+		if(flag.team != src.team)
+			for(var/mob/M in player_list)
+				if (M.z == src.z)
+					M << "<span class='userdanger'>[user.real_name] has captured /the [src], scoring a point for [team] team! They now have [points]/[points_to_win] points!</span>"
+		if(points == points_to_win)
+			victory()
+
+
+
+/obj/machinery/capture_the_flag/proc/victory()
+	for(var/mob/M in player_list)
+		if (M.z == src.z)
+			M << "<span class='narsie'>[team] team wins!</span>"
```

Comment:
MAKE A NEW SPAN CLASS YOU NERD DON'T USE THIS ONE
HECK, PASS THE WINNING TEAM'S COLOR DOWN TO THIS SO YOU CAN MAKE IT ANY COLOR